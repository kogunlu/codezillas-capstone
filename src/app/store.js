import { configureStore } from '@reduxjs/toolkit';
import counterReducer from '../features/counter/counterSlice';
import bookingReducer from '../features/booking/bookingSlice';
import userReducer from '../features/user/userSlice';

export const store = configureStore({
  reducer: {
    counter: counterReducer,
    booking: bookingReducer,
    user: userReducer,
  },
});
