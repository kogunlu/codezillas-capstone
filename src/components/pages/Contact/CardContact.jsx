import React from 'react'
import { useDispatch  } from 'react-redux';
import swal from 'sweetalert';



function CardContact() {

    

    const dispatch = useDispatch()
    
        const handleContactType = (e) => {
            setContactType(e.target.value)
            dispatch()
        }

   function handleClick(){
        if(preference === ""){
            swal("Error!", "Please select an answer for this question!!");
        }else{
            dispatch((preference))
            btnClick()
        }
    } 

  return (
    <div className=' h-96 w-full md:w-5/6 lg:w-4/6 px-2 md:px-10 py-5 flex  justify-between items-start   flex-wrap'>

        <div className='h-auto mb-6'>
            <h2 className='text-xl font-medium'>
                Type of contact
            </h2>
        </div>

        <div className='w-full h-auto py-5 md:py-0 flex flex-col justify-center items-start gap-1 '>
            <div className='mb-2'>
                <input 
                type="radio" 
                value={contactType}
                onChange={handleChange}
                checked= {preference === 'service'}
                /> I have a question about the service.
            </div>
                
            <div className='mb-2'>
                <input 
                type="radio" 
                value={contactType}
                onChange={handleChange} 
                checked= {preference === 'support'}
                /> I’m a registered client and I need support.
            </div>

            <div className='mb-2'>
                <input
                type="radio" 
                value={contactType}
                onChange={handleChange}
                checked= {preference === 'joining'}
                /> I’m a counselor interested in joining.
            </div>

            <div className='mb-2'>
                <input
               
                type="radio" 
                value={contactType} 
                onChange={handleChange}
                checked= {preference === 'counselorSupport'}
                /> I’m a registered counselor and I need support.
            </div>    
                
            <div className='mb-2'>
                <input
           
                type="radio" 
                value={contactType}
                onChange={handleChange}
                checked= {preference === 'business'}
                /> I have a business-related inquiry.
            </div>

            <div className='mb-2'>
                <input
                type="radio" 
                value={contactType}
                onChange={handleChange}
                checked= {preference === 'healingOnline'}
                /> I’m interested in Healing Online for my organization.
            </div>

            <div>
                <input
                type="radio" 
                value={contactType}
                onChange={handleChange}
                checked= {preference === 'billing'}
                /> I have a billing related question.
            </div>
        </div>

      
    </div>
  )
}

export default CardContact