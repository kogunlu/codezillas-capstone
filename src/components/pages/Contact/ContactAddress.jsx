import React from "react";
import ContactAdressImg from "./images/ContactAdressImg.png";


function ContactAddress () {
   return ( 
   <div>
    <div className="relative mt-28 ml-15 m-10">
        <img className="h-56 w-96 md:h-auto md:w-auto "src={ContactAdressImg} alt="adress" /> 
      <span className="absolute top-3 left-3 w-4/6 pl-5 pt-3">
              <p className="font-medium text-xs md:text-lg text-start mb-1">
                Find Us At: 
              </p>
             <div className="text-gray-500 text-xs md:text-lg">
              <p>Nergiz Plaza</p>
              <p>3rd Floor</p>
              <p>Bakhtiyari Street 40m</p>
              <p>Erbil, Iraq</p>
              <p>44001</p>
              </div>
            </span>
            </div>
            </div>
      
   )
}
export default ContactAddress ;