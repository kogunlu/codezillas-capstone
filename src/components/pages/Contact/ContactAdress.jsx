import React from "react";
import ContactAdressImg from "./images/ContactAdressImg.png";


function ContactAdress () {
   return ( 
   <div>
    <div className="relative"><img
        className=" items-start lg:justify-center rounded mx-auto py-7  "
        src={ContactAdressImg}
        alt="meditation"
      /> 
      <span className="absolute top-3 left-3 w-4/6">
              <p className="font-medium text-sm md:text-lg text-start">
                Find Us At: Nergiz Plaza 3rd Floor
              </p>
            </span>
            </div>
            </div>
      
   )
}
export default ContactAdress ;



