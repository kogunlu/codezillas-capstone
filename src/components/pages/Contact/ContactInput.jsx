import {useState} from 'react'
import {useNavigate} from "react-router-dom"
import { v4 as uuidv4 } from 'uuid';
import { doc, setDoc } from "firebase/firestore"; 
import db from "../../../db/firebase.config"

function ContactInput(name, email, details){
    
  const [name, setName] = useState("")
  const [email, setEmail] = useState("")
  const [details, setDetails] = useState("")
  const navigate = useNavigate()

  async function sendData(obj){
    
    await setDoc(doc(db, "subscription-list", uuidv4()), {
      input: obj
    });

  }

 

  function onChange(){
    sendData(name, email, details).then(setName, setEmail, setDetails("")).then(navigate('/thanks-contacts'))
  }
 
    return {
        name, email, details 
       render: (
       <form className=' w-full md:w-1/4 lg:w-4/6 px-2 md:px-8 sm:text-small  '>
        <span className='mb-3 '>
            <label htmlFor='name' className='font-medium'>Full Name:
            <input type="name" placeholder='Enter your full name here...' value={nameValue} onChange={handleNameChange} className=' box-border w-full border rounded-md
      bg-transparent border-slate-300 p-3 mt-2 mb-2 shadow-md focus:outline-none'/> 
            </label>
        </span>
        <span className='mb-3 '>
            <label htmlFor='email' className='font-medium'>Email:
            <input type="email" placeholder='Enter your email address here...' value={email} onChange={(e) => setEmail(e.target.value)} className='w-full border rounded-md
             border-slate-300 p-3 mt-2 mb-2 shadow-md focus:outline-none'/>
            </label>
        </span>
        <span className=''>
            <label htmlFor='details' className='font-medium'>Details:
            <textarea type="details" rows={6} placeholder='Enter your details here...' value={details} onChange={(e) => setDetails(e.target.value)} className='w-full border rounded-md
             border-slate-300 p-3 mt-2 shadow-md focus:outline-none'  style={{resize: "none"}}/>
            </label>
        </span>
        <div className='flex items-center w-2/3 h-1/7 mb-12 mt-5 focus:outline-none'>
            <input 
            type="button" 
            value="SUBMIT" 
            className='py-2 px-5 w-6/12 md:w-5/12 lg:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-medium rounded'
            onClick={() => onChange()} />
          
        </div> 
       </form>
    )}
}

export default ContactInput