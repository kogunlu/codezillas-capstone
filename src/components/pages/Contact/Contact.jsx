/* eslint-disable no-unused-vars */
import React, {useState} from "react";
import {useNavigate} from "react-router-dom";
import { doc, setDoc } from "firebase/firestore"; 
import { useDispatch  } from 'react-redux';
import db from "../../../db/firebase.config";
import contact from "./images/Contact.png";
import ContactAddress from "./ContactAddress";




function Contact () {
    const [contactType, setContactType] = useState("") 
    const [name,setName] = useState("")
    const [email,setEmail] = useState("")
    const [details, setDetails] = useState("")
    const dispatch = useDispatch()
    const navigate = useNavigate()

    const handleContactChange = e => {setContactType(e.target.value)}
    const handleNameChange = e => {setName(e.target.value)}
    const handleEmailChange = e => {setEmail(e.target.value)}
    const handleDetailsChange = e => {setDetails(e.target.value)}
    
   
   
    
      
        async function sendData(contactTypeValue1, nameValue1, emailValue1, detailsValue1){
          
          await setDoc(doc(db, "contactUs-list", emailValue1), {
            contactType: contactTypeValue1,
            name: nameValue1,
            email: emailValue1,
            details: detailsValue1
          })
          .then(handleContactChange, handleNameChange, handleEmailChange, handleDetailsChange, ("")).then(navigate('/thanks-contacts'))
        }
         
        


   return ( 
   <div className="">
    <div className="pl-2 py-10 md:pl-10 ">
        <h1 className="text-sm md:text-5xl xs:text-left ">SEND US YOUR REQUEST!</h1>
        <p className="text-gray-500  mt-2 text-xs md:text-3xl ">Do you have a question, concern, idea, feedback, or problem? If you need assistance,
             please fill out the <br/> form below and we’d be happy to help!</p>
    </div>
    <div className="flex ">
  
    <div>

<div className=' h-auto md:h-auto md:text-xl lg:w-4/6 px-2 md:px-10 py-3 md:py-3 flex  justify-between items-start   flex-wrap'>

    <div className='h-auto mb-6'>
        <h2 className='xs:text-xs md:text-xl font-medium'>
            Type of contact
        </h2>
    </div>

    <div className=' py-0 md:py-0 flex flex-col justify-center items-start gap-1 text-xs md:text-xl '>
        <div className='mb-2'>
            <input 
            name="contactType"
            type="radio" 
            value="I have a question about the service."
            onChange={handleContactChange}
           
            /> I have a question about the service.
        </div>
            
        <div className='mb-2'>
            <input 
            name="contactType"
            type="radio" 
            value="I’m a registered client and I need support."
            onChange={handleContactChange} 
            
            /> I’m a registered client and I need support.
        </div>

        <div className='mb-2'>
            <input
            name="contactType"
            type="radio" 
            value=" I’m a counselor interested in joining."
            onChange={handleContactChange}
          
            /> I’m a counselor interested in joining.
        </div>

        <div className='mb-2'>
            <input
           name="contactType"
            type="radio" 
            value="I’m a registered counselor and I need support." 
            onChange={handleContactChange}
           
            /> I’m a registered counselor and I need support.
        </div>    
            
        <div className='mb-2'>
            <input
            name="contactType"
            type="radio" 
            value="I have a business-related inquiry."
            onChange={handleContactChange}
            
            /> I have a business-related inquiry.
        </div>

        <div className='mb-2'>
            <input
            name="contactType"
            type="radio" 
            value="I’m interested in Healing Online for my organization."
            onChange={handleContactChange}
           
            /> I’m interested in Healing Online for my organization.
        </div>

        <div>
            <input
            name="contactType"
            type="radio" 
            value="I have a billing related question."
            onChange={handleContactChange}
          
            /> I have a billing related question.
        </div>
    </div>
</div>
    </div>
    
    <div><img
        className=" items-start lg:justify-center rounded mx-auto py-7  w-96 md:h-auto md:w-auto "
        src={contact} 
        alt="contact"
      /> </div> 
   </div>
      <div className="flex ">
     
      
  
      
         <form className=' w-full md:w-2/4 lg:w-4/6 px-2 md:px-8 text-xs md:text-lg  '>
          <span className='mb-3 '>
              <label htmlFor='name' className='font-medium text-xs md:text-lg'>Full Name:
              <input type="name" placeholder='Enter your full name here...' value={name} onChange={handleNameChange} className=' box-border w-full border rounded-md
        bg-transparent border-slate-300 p-3 mt-2 mb-2 shadow-md focus:outline-none'/> 
              </label>
          </span>
          <span className='mb-3 '>
              <label htmlFor='email' className='font-medium'>Email:
              <input type="email" placeholder='Enter your email address here...' value={email} onChange={handleEmailChange} className='w-full border rounded-md
               border-slate-300 p-3 mt-2 mb-2 shadow-md focus:outline-none'/>
              </label>
          </span>
          <span className=''>
              <label htmlFor='details' className='font-medium'>Details:
              <textarea type="details" rows={6} placeholder='Enter your details here...' value={details} onChange={handleDetailsChange} className='w-full border rounded-md
               border-slate-300 p-3 mt-2 shadow-md focus:outline-none'  style={{resize: "none"}}/>
              </label>
          </span>
          <div className='flex items-center w-2/3 h-1/7 mb-12 mt-5 focus:outline-none'>
              <input 
              type="button" 
              value="SUBMIT" 
              className='py-2 px-5 w-6/12 md:w-5/12 lg:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-medium rounded '
              onClick={() => { 
                sendData(contactType, name, email, details); 
                }} />
    
          </div> 
         </form>
      
  
        <ContactAddress />
      </div>
   </div>
   )
}
export default Contact;