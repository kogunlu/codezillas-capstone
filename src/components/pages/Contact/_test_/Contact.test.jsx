import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import Contact from '../Contact';
import { store } from '../../../../app/store';

const mockedUsedNavigate = jest.fn();

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

it('renders correctly ', async () => {
  const errorComponent = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Contact />
        </BrowserRouter>
      </Provider>
    )
    .toJSON();
  expect(errorComponent).toMatchSnapshot();
});
