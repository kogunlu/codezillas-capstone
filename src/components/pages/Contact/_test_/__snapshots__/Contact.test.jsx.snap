// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly  1`] = `
<div
  className=""
>
  <div
    className="pl-2 py-10 md:pl-10 "
  >
    <h1
      className="text-sm md:text-5xl xs:text-left "
    >
      SEND US YOUR REQUEST!
    </h1>
    <p
      className="text-gray-500  mt-2 text-xs md:text-3xl "
    >
      Do you have a question, concern, idea, feedback, or problem? If you need assistance, please fill out the 
      <br />
       form below and we’d be happy to help!
    </p>
  </div>
  <div
    className="flex "
  >
    <div>
      <div
        className=" h-auto md:h-auto md:text-xl lg:w-4/6 px-2 md:px-10 py-3 md:py-3 flex  justify-between items-start   flex-wrap"
      >
        <div
          className="h-auto mb-6"
        >
          <h2
            className="xs:text-xs md:text-xl font-medium"
          >
            Type of contact
          </h2>
        </div>
        <div
          className=" py-0 md:py-0 flex flex-col justify-center items-start gap-1 text-xs md:text-xl "
        >
          <div
            className="mb-2"
          >
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value="I have a question about the service."
            />
             I have a question about the service.
          </div>
          <div
            className="mb-2"
          >
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value="I’m a registered client and I need support."
            />
             I’m a registered client and I need support.
          </div>
          <div
            className="mb-2"
          >
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value=" I’m a counselor interested in joining."
            />
             I’m a counselor interested in joining.
          </div>
          <div
            className="mb-2"
          >
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value="I’m a registered counselor and I need support."
            />
             I’m a registered counselor and I need support.
          </div>
          <div
            className="mb-2"
          >
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value="I have a business-related inquiry."
            />
             I have a business-related inquiry.
          </div>
          <div
            className="mb-2"
          >
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value="I’m interested in Healing Online for my organization."
            />
             I’m interested in Healing Online for my organization.
          </div>
          <div>
            <input
              name="contactType"
              onChange={[Function]}
              type="radio"
              value="I have a billing related question."
            />
             I have a billing related question.
          </div>
        </div>
      </div>
    </div>
    <div>
      <img
        alt="contact"
        className=" items-start lg:justify-center rounded mx-auto py-7  w-96 md:h-auto md:w-auto "
        src="Contact.png"
      />
       
    </div>
  </div>
  <div
    className="flex "
  >
    <form
      className=" w-full md:w-2/4 lg:w-4/6 px-2 md:px-8 text-xs md:text-lg  "
    >
      <span
        className="mb-3 "
      >
        <label
          className="font-medium text-xs md:text-lg"
          htmlFor="name"
        >
          Full Name:
          <input
            className=" box-border w-full border rounded-md bg-transparent border-slate-300 p-3 mt-2 mb-2 shadow-md focus:outline-none"
            onChange={[Function]}
            placeholder="Enter your full name here..."
            type="name"
            value=""
          />
        </label>
      </span>
      <span
        className="mb-3 "
      >
        <label
          className="font-medium"
          htmlFor="email"
        >
          Email:
          <input
            className="w-full border rounded-md border-slate-300 p-3 mt-2 mb-2 shadow-md focus:outline-none"
            onChange={[Function]}
            placeholder="Enter your email address here..."
            type="email"
            value=""
          />
        </label>
      </span>
      <span
        className=""
      >
        <label
          className="font-medium"
          htmlFor="details"
        >
          Details:
          <textarea
            className="w-full border rounded-md border-slate-300 p-3 mt-2 shadow-md focus:outline-none"
            onChange={[Function]}
            placeholder="Enter your details here..."
            rows={6}
            style={
              Object {
                "resize": "none",
              }
            }
            type="details"
            value=""
          />
        </label>
      </span>
      <div
        className="flex items-center w-2/3 h-1/7 mb-12 mt-5 focus:outline-none"
      >
        <input
          className="py-2 px-5 w-6/12 md:w-5/12 lg:w-3/12 hover:bg-cyan-200 bg-cyan-400 font-medium rounded "
          onClick={[Function]}
          type="button"
          value="SUBMIT"
        />
      </div>
    </form>
    <div>
      <div
        className="relative mt-28 ml-15 m-10"
      >
        <img
          alt="adress"
          className="h-56 w-96 md:h-auto md:w-auto "
          src="ContactAdressImg.png"
        />
        <span
          className="absolute top-3 left-3 w-4/6 pl-5 pt-3"
        >
          <p
            className="font-medium text-xs md:text-lg text-start mb-1"
          >
            Find Us At:
          </p>
          <div
            className="text-gray-500 text-xs md:text-lg"
          >
            <p>
              Nergiz Plaza
            </p>
            <p>
              3rd Floor
            </p>
            <p>
              Bakhtiyari Street 40m
            </p>
            <p>
              Erbil, Iraq
            </p>
            <p>
              44001
            </p>
          </div>
        </span>
      </div>
    </div>
  </div>
</div>
`;
